plugins/linguist/src/components/LinguistCard/LinguistCard.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Box,
+  Chip,
+  Tooltip,
+  Typography,
+  makeStyles,
+  Grid,
+  IconButton,
+  useTheme,
+} from '@material-ui/core';
+import { InfoCard, Progress } from '@backstage/core-components';
+import Alert from '@material-ui/lab/Alert';
+import { DateTime } from 'luxon';
+import React, { useCallback } from 'react';
+import slugify from 'slugify';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { useLanguages } from '../../hooks';
+import CachedIcon from '@material-ui/icons/Cached';
+import { alertApiRef, useApi } from '@backstage/core-plugin-api';
+import { linguistApiRef } from '../../api';
+import { usePermission } from '@backstage/plugin-permission-react';
+import { linguistUpdateLanguagesPermission } from '@backstage/plugin-linguist-common';
+
+const useStyles = makeStyles(theme => ({
+  infoCard: {
+    marginBottom: theme.spacing(3),
+    '& + .MuiAlert-root': {
+      marginTop: theme.spacing(3),
+    },
+  },
+  barContainer: {
+    height: theme.spacing(2),
+    marginBottom: theme.spacing(3),
+    borderRadius: '4px',
+    backgroundColor: 'transparent',
+    overflow: 'hidden',
+  },
+  bar: {
+    height: '100%',
+    position: 'relative',
+  },
+  languageDot: {
+    width: '10px',
+    height: '10px',
+    borderRadius: '50%',
+    marginRight: theme.spacing(1),
+    display: 'inline-block',
+  },
+  label: {
+    color: 'inherit',
+  },
+}));
+
+export const LinguistCard = () => {
+  const classes = useStyles();
+  const theme = useTheme();
+  const { entity } = useEntity();
+  const linguistApi = useApi(linguistApiRef);
+  const alertApi = useApi(alertApiRef);
+  const { items, loading, error } = useLanguages(entity);
+  let barWidth = 0;
+  const { loading: loadingPermission, allowed: canUpdateLanguages } =
+    usePermission({ permission: linguistUpdateLanguagesPermission });
+  const refreshLanguages = useCallback(async () => {
+    linguistApi.refreshLanguages(entity);
+    alertApi.post({ message: 'Languages refresh scheduled', severity: 'info' });
+  }, [linguistApi, alertApi, entity]);
+
+  const refreshAction = !loadingPermission && canUpdateLanguages && (
+    <IconButton
+      aria-label="Refresh Languages"
+      title="Refresh Languages"
+      onClick={refreshLanguages}
```

Comment:
This is another spot I could use some advice on. Generating the language data is an expensive operation. I wasn't sure the best way to `debounce` this or prevent people going click happy here. I do have this tied into the permissions so ideally those that would click this would have some foresight

Reply:
refresh on entities in the catalog is just "update a timestamp to now()" and then the regular refresh loops continue going about their happy ways

Reply:
I've taken out the refresh option along with the related permissions. This simplifies things so that I can focus on getting the processing right. I'll add it back in a later PR